### PR TITLE
Discord new guild member trigger

### LINKED ISSUE #2238: [TRIGGER] Discord on guild member add
**Describe the event source. What app is this for, and what event does the trigger correspond to?**

Would love to have a trigger for when a member is added to a guild.

**Please provide a link to the relevant API docs for the specific service / operation this trigger is tied to.**

https://discord.com/developers/docs/topics/gateway#guild-member-add

### PR DESCRIPTION
**Pull request by andrewjschuang**
New source / trigger:

- [X] New Guild Member

Resolves #2238

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/discord_bot/sources/new-guild-member/new-guild-member.mjs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  async run({ $ }) {
+    const guildMembers = this._getGuildMemberIDs();
+    const { guildId } = this;
+    const params = {
+      limit: 100,
+    };
+    let anyNewMember = false;
+
+    while (true) {
```
Comment (vellames-turing) [inline components/discord_bot/sources/new-guild-member/new-guild-member.mjs:31]: I don't get why you need a `while(true)` here.  Could you please explain to me?

**Review by vellames-turing — CHANGES_REQUESTED** [review]
Hey just one minor and one question

File: components/discord_bot/sources/new-guild-member/new-guild-member.mjs (hunk 1 above)
Comment (andrewjschuang) [reply components/discord_bot/sources/new-guild-member/new-guild-member.mjs:31]: > I don't get why you need a while(true) here. Could you please explain to me?

It's due to pagination. When [listing guild members](https://discord.com/developers/docs/resources/guild#list-guild-members), the `after` param is **the highest user id in the previous page, inclusive**. So the only way to know if the final member was reached is if the response contains only one member.

Comment (vellames-turing) [reply components/discord_bot/sources/new-guild-member/new-guild-member.mjs:31]: What if we store this after to not repeat the request with the previous users, is it possible? Maybe you can use a `do/while` until `after` to be `null`? Normally the `after` points to the last record on the page. So you can always save the last record as `after`. Does it make sense for you?

Comment (andrewjschuang) [reply components/discord_bot/sources/new-guild-member/new-guild-member.mjs:31]: Yeah, I thought about that but members with a lower `id` can be added to a guild after the run, so I think in those cases they would be skipped

**Review by vellames-turing — APPROVED** [review]

**Comment by andrewjschuang** [comment]
Hey @DilanAthukorala, are you selecting the prop via `prop options`? Or are you typing in the `guild name`?
It seems that the request is sending the `guild name`, but it should be the [`snowflake guildId`](https://discord.com/developers/docs/reference#snowflakes)
>     "guild_id": [
>       "Value \"PDEventGuild\" is not snowflake."
>     ]

**Pull request merged**